src/lib/components/table/table-column-settings/dialog/table-settings-dialog.component.ts

```diff
@@ -32,7 +32,35 @@ export class TableSettingsDialogComponent implements OnInit
      */
     public ngOnInit():void
     {
-        this._columns = this.data.columns;
-        this._selectedColumns = this.data.selectedColumns;
+        this._selectedColumns = this.data.selectedColumns.slice();
+        this._columns = this._sort(this.data.columns);
+    }
+
+
+    public _onDrop(event:CdkDragDrop<Array<MatColumnDef>>):void
+    {
+        moveItemInArray(this._selectedColumns, event.previousIndex, Math.min(event.currentIndex, this._selectedColumns.length));
+
+        this._columns = this._sort(this._columns);
+    }
+
+    public _sort(cols:Array<MatColumnDef>):Array<MatColumnDef>
+    {
+        let selectedList:Array<MatColumnDef> = this._selectedColumns.map((key:string) =>
+        {
+            return cols.find((col:MatColumnDef) => col.name === key);
+        });
+
+        let unselectedList:Array<MatColumnDef> = [];
+
+        cols.forEach((col:MatColumnDef) =>
+        {
+            if(!this._selectedColumns.includes(col.name))
+            {
+                unselectedList.push(col);
+            }
+        });
```

Comment:
Hier kannst du nun Array.filter nutzen:
```suggestion
        let unselectedList:Array<MatColumnDef> = cols.filter((col:MatColumnDef) =>
        {
            return !this._selectedColumns.includes(col.name));
        });
```